### PR TITLE
chore: e2e tests

### PR DESCRIPTION
**Pull request by eh-am**
It uses https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript for a nice DLS for testing the ci setup against different languages/frameworks (running in Docker).

Example from nodejs test
```bash
pyroscope-ci exec --noUpload --export -- docker run --link $PYROSCOPE_PROXY_ADDRESS --env PYROSCOPE_ADHOC_SERVER_ADDRESS $IMAGE_NAME yarn test
exists pyroscope-ci-output/pyroscope.tests.cpu.json
```

Adding new tests is straightforward (from the `README`):
```
1. Create a `Dockerfile`
2. Add a test (see `e2e_test.go` for reference)
3. Write a `testscript.txtar` file
```

### CODE REVIEW

**Comment by eh-am** [comment]
As an example on how to add new examples see the following commit https://github.com/pyroscope-io/ci/pull/34/commits/d698e31a624b50cf1e07b425450387da63b485c0

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -17,9 +17,29 @@ jobs:
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: asdf_install
+        uses: asdf-vm/actions/install@v1
+      - run: make install-dev-tools
+      # Make asdf aware of globally installed binaries
+      - run: asdf reshim
+      - run: make test
+  e2e-tests:
```
Comment (petethepig) [inline .github/workflows/ci.yaml:32]: I'd move this to a separate yml file, so that it can run in parallel

**Review by petethepig — APPROVED** [review]
Looks good overall, I like the idea!

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (eh-am) [reply .github/workflows/ci.yaml:32]: I believe that's how it works currently 
image
https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow

**Pull request merged**